server/ingest/handler/ingest_handler.go

```diff
@@ -107,3 +113,36 @@ func (h *IngestHandler) IndexContent(ctx context.Context, data []byte) error {
 
 	return nil
 }
+
+const maxAnnounceSize = 512
+
+func (h *IngestHandler) Announce(ctx context.Context, data io.Reader) error {
+	bytes, err := io.ReadAll(io.LimitReader(data, maxAnnounceSize))
+	if err != nil {
+		return err
+	}
+	an := dtsync.Message{}
+	if err := json.Unmarshal(bytes, &an); err != nil {
+		return err
+	}
+	// todo: support mulitple multiaddrs?
+	if len(an.Addrs) > 1 {
+		return fmt.Errorf("must specify 1 location to fetch on direct announcments")
+	}
+	// todo: require auth?
+
+	pid, err := peer.AddrInfoFromP2pAddr(an.Addrs[0])
+	if err != nil {
+		return err
+	}
+
```

Comment:
```suggestion
	allow, err := h.registry.Authorized(pid)
	if err != nil {                                                                                                                                                      
		err = fmt.Errorf("error checking if peer allowed: %w", err)
		return v0.NewError(err, http.StatusInternalServerError)
	}
	if !allow {
		return v0.NewError(errors.New("not authorized to announce"), http.StatusForbidden)
	}
```
This includes the HTTP status with the error message.